File: backend/internal/model/remuneracao.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// Remuneracao representa o registro de remuneracao da integradora por lote.
type Remuneracao struct {
	ID             uuid.UUID  `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	LoteID         uuid.UUID  `gorm:"type:uuid;not null" json:"lote_id" validate:"required"`
	Valor          float64    `gorm:"type:numeric(12,2);not null" json:"valor" validate:"required"`
	Tipo           *string    `gorm:"type:varchar(30)" json:"tipo,omitempty" validate:"omitempty,oneof=por_kg por_ave por_iep outro"`
	DataPagamento  *time.Time `gorm:"type:date" json:"data_pagamento,omitempty"`
	Observacao     *string    `gorm:"type:text" json:"observacao,omitempty"`
	CreatedAt      time.Time  `gorm:"not null;default:now()" json:"created_at"`

	// Relacionamentos
	Lote Lote `gorm:"foreignKey:LoteID" json:"-"`
}

func (Remuneracao) TableName() string {
	return "remuneracoes"
}

func (r *Remuneracao) BeforeCreate(tx *gorm.DB) error {
	if r.ID == uuid.Nil {
		r.ID = uuid.New()
	}
	return nil
}

File: flutter_app/lib/features/financeiro/domain/entities/remuneracao.dart
/// Registro de remuneracao de um lote.
///
/// Representa o pagamento recebido pela integradora, incluindo
/// valor, tipo de remuneracao, data de pagamento e observacao.
class Remuneracao {
  final String id;
  final String loteId;
  final double valor;
  final String? tipo;
  final String? dataPagamento;
  final String? observacao;
  final String createdAt;

  const Remuneracao({
    required this.id,
    required this.loteId,
    required this.valor,
    this.tipo,
    this.dataPagamento,
    this.observacao,
    required this.createdAt,
  });

  factory Remuneracao.fromJson(Map<String, dynamic> json) {
    return Remuneracao(
      id: json['id'] as String,
      loteId: json['lote_id'] as String,
      valor: (json['valor'] as num?)?.toDouble() ?? 0.0,
      tipo: json['tipo'] as String?,
      dataPagamento: json['data_pagamento'] as String?,
      observacao: json['observacao'] as String?,
      createdAt: json['created_at'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'lote_id': loteId,
      'valor': valor,
      if (tipo != null) 'tipo': tipo,
      if (dataPagamento != null) 'data_pagamento': dataPagamento,
      if (observacao != null) 'observacao': observacao,
      'created_at': createdAt,
    };
  }

  /// Label legivel para o tipo de remuneracao.
  static String tipoLabel(String? tipo) {
    if (tipo == null) return 'Nao informado';
    const labels = {
      'por_kg': 'Por kg',
      'por_ave': 'Por ave',
      'por_iep': 'Por IEP',
      'outro': 'Outro',
    };
    return labels[tipo] ?? tipo;
  }
}
